File: Código/Backend/PharmaGo.Domain/Entities/Product.cs
using PharmaGo.Exceptions;
using System.Collections.Generic;

namespace PharmaGo.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public Pharmacy? Pharmacy { get; set; }

        public void ValidOrFail()
        {
            if (string.IsNullOrEmpty(Code) || string.IsNullOrEmpty(Name) || Price <= 0 || string.IsNullOrEmpty(Description))
            {
                throw new InvalidResourceException("The Drug was not correctly created.");
            }
            if (Name.Length > 30 || Description.Length > 70 || Code.Length != 5)
            {
                throw new InvalidResourceException("The Drug format is incorrect");
            }
        }
    }
}

File: Código/Frontend/src/app/interfaces/product.ts
export interface Product {
    id: number;
    code: string;
    name: string;
    price: number;
    description: string;
    pharmacy: {
        id: number;
        name: string;  
      };
  }

  export class ProductRequest {
    code: string;
    name: string;
    price: number;
    description: string;
    pharmacyName: string = "";

    constructor(code: string, name: string, price: number, description: string, pharmacyName: string){
      this.code = code;
      this.name = name;
      this.price = price;
      this.description = description;
      this.pharmacyName = pharmacyName;
    }
  }
  
  export class UpdateProductRequest {
    code: string;
    name: string;
    price: number;
    description: string;
  

    constructor(code: string, name: string, price: number, description: string){
      this.code = code;
      this.name = name;
      this.price = price;
      this.description = description;
    }
  }
